src/components/Layerboard.vue

```diff
@@ -88,6 +109,10 @@
         }
       }
 
+      if (this.$config.dataSources) {
+        this.$controller.dataManager.fetchData();
```

Comment:
Is this operation async? If so, I wonder if we should be using async... await to ensure that we're not seeing race conditions later in the code.

Reply:
yes, fetchData() is an enormous function that loops ajax calls. Happy to look at it with you and try to use promises to make it more reliable and efficient some time